libraries/Microsoft.Bot.Connector/Authentication/Government/MicrosoftGovernmentAppCredentials.cs

```diff
@@ -62,5 +67,72 @@ public override string OAuthEndpoint
         {
             get { return GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl; }
         }
+
+        protected override Lazy<IAuthenticator> BuildIAuthenticator()
+        {
+            return new Lazy<IAuthenticator>(
+                () =>
+                {
+                    var credential = new ClientCredential(MicrosoftAppId, MicrosoftAppPassword);
+
+                    IAuthenticator Make(string authority, string scope)
+                        => new AdalAuthenticator(
+                            credential,
+                            new OAuthConfiguration() { Authority = authority, Scope = scope },
+                            this.CustomHttpClient,
+                            this.Logger);
+
+                    if (OAuthScope == GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope)
+                    {
+                        var optionOld = Make(GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl, GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope);
+                        var optionNew = Make("https://login.microsoftonline.us/botframework.onmicrosoft.us", "https://api.botframework.onmicrosoft.us");
+
+                        return new Authenticators(optionOld, optionNew);
```

Comment:
keep optionNew as your first argument. almost all new bots going forward will be in that new tenant and hence trying the new one first will make sure that the for loop doesnt execute more than one iteration for most new bots